sdk/core/core/inc/az_span.h

```diff
@@ -260,6 +260,34 @@ AZ_NODISCARD az_result az_span_to_uint64(az_span span, uint64_t* out_number);
  */
 AZ_NODISCARD az_result az_span_to_uint32(az_span span, uint32_t* out_number);
 
+/**
+ * @brief az_span_find searches for `target` in `source`, returning an #az_span if it finds it.
+ *
+ * @param[in] source The #az_span containing the ASCII digits to be searched on.
+ * @param[in] target The #az_span containing the token to be searched in `source`.
+ * @param[out] out_span The #az_span pointing to the first occurrence of `target` in `source`, if it is found.
+ * @return An #az_result value indicating the result of the operation.
+ *          #AZ_OK if `target` is found in `source`
+ *          #AZ_ERROR_EOF if `target` is not found in `source`
+ * is found within the span.
+ */
+AZ_NODISCARD az_result az_span_find(az_span source, az_span target, az_span* out_span);
+
+/**
+ * @brief az_span_token_next is a string tokenizer for az_span.
+ *
+ * @param[in] source The #az_span containing the ASCII digits to be searched on.
+ * @param[in] delimiter The #az_span containing the delimiter to "split" `source` into tokens.
+ * @param[out] out_token The #az_span pointing to the token delimited by the beginning of `source` 
+ *                       up to the first occurrence of `delimiter`, or the end of `source` 
+ *                       if `delimiter` is not found.
+ * @param[out] out_source The #az_span pointing to the next token in `source`, starting after the occurrence of `delimiter`.
```

Comment:
Don't you want this to be the remainder of the source (all bytes after the delimiter)? If so, I'd call it out_remainder.
This way, I can call this function in a loop passing in for source the previous result in out_remainder.

Reply:
That's exactly how this function works right now.
I'll change out_source to out_remainder.

Reply:
Your doc says az_span points to the next "token" in source but it's not the next token; it is the full remainder of the source (which may contain multiple more tokens in it). This is why I'm suggesting the parameter change and a doc change.